### ISSUE TITLE
nemo crashing with segfault

### ISSUE THREAD

**Issue by blackdot** [linuxmint/nemo#2019]
* Nemo version : 4.0.6
 * Issue with windowed nemo
 * Distribution - Mint 19.1
 * Graphics hardware GeForce GT 710B / nouveau
 * 64 bit

**Issue**
When opening nemo in home (/home/user/) folder or any main folder synchronized with nextcloud (/home/user/DocumentsCloud/ , /home/user/WorkCloud/ , ...), nemo crashed with segfault.
Any subfolder of nextcloud synchronized folders (/home/user/WorkCloud/A/ ) works fine (nemo is not crashing).

dmesg:
`[16474.559415] nemo[25060]: segfault at 10 ip 00007fe5fade945c sp 00007ffcb0797960 error 4 in libglib-2.0.so.0.5600.3[7fe5fadb3000+113000]
[16474.559423] Code: 4a 27 06 00 48 8d 3d d1 e7 05 00 48 83 ec 08 e8 8a d6 01 00 31 c0 48 83 c4 08 c3 0f 1f 00 55 53 48 89 fd 48 89 f3 48 83 ec 08 <8b> 77 10 48 8b 7f 08 e8 08 25 04 00 48 63 75 14 48 ba 00 60 d7 1d`

.xsession_errors
`(caribou:10368): GLib-GObject-CRITICAL **: 17:12:03.297: g_object_unref: assertion 'G_IS_OBJECT (object)' failed`

**Steps to reproduce**
1) Opening nemo which by defaults displays /home/user/ folder. Then it crash immediately.
2a) Executing "nemo /tmp" works fine.
2b) After going to ~ directory, nemo crash.

**Expected behaviour**
Nemo is not crashing.

**Other information**
gdb is here: 
[https://pastebin.com/raw/ECXDk7Ca]

**Comment by okaestne** [comment]
My issue #2018 seems to be directly related. I also got two folders which are in sync with my nextcloud.
(Segfaults at the same position)

**Comment by blackdot** [comment]
Oh yes, looks like we are hitting same issue.
When I changed the view to compact (using command from your issue), I am able to see the content of all folders without crashing.
Can you confirm, if this is happening to you only to folders which are synced by nextcloud (and parent directory)?

Small addition, I was using owncloud-client and nextcloud-server. Today I migrated client from owncloud to nextcloud and it is still same - crashing.

**Comment by mtwebster** [comment]
Also, can you share your nemo settings:

```
dconf dump /org/nemo/ > nemo.txt
```

I've created an owncloud account on a free server, and set it up using the linux client, and it seems to work fine for me in the list view, so I need to see if there are any differences between us.

Thanks

**Comment by okaestne** [comment]
> Are you guys able to tell me what columns you have enabled in the list view that's crashing?

None for this particular folder. I already tried removing the "Date Modified " column in the Nemo preferences as well as removing all nemo-* metadata without success.
To answer you question: only Name, Size and Type right now.
 
> Can you both confirm this _only_ happens in the list view, not icon or compact?

For me list view only.

> Also, can you share your nemo settings:
> 
> ```
> dconf dump /org/nemo/ > nemo.txt
> ```

https://gist.github.com/okaestne/42c93ed2f179f49d5e474510df9a7cb7

> I've created an owncloud account on a free server, and set it up using the linux client, and it seems to work fine for me in the list view, so I need to see if there are any differences between us.
> 
> Thanks

For me all synced folders and their parent folders are affected.
I'm using the Nextcloud client (2.5.1) installed via the Nextcloud repositories. (so not the AppImage version if this matters..)

**Comment by mtwebster** [comment]
Same behavior with 2.5.1 appimage (and using those nemo preferences).  Do you store Windows files in these folders also?  I'll set the client up in Windows and store some files from there as well, if so.

**Comment by okaestne** [comment]
What exactly do you mean with "Windows files"? I got different sync folders, for music, photos, documents.. All the same issue.

**Comment by mtwebster** [comment]
I mean do you sync between different computers/operating systems?  Or do you only use linux?

**Comment by okaestne** [comment]
> Can you open a terminal in those folders and `ls -la` to see if there are any peculiarities in folders that are affected vs those that aren't? (such as different owners, permissions, etc..)

Everything owned by me with 755 / 664 permissions. Same issue when opened in Nemo root mode, so permissions can't by the cause.

> Sorry for excessive requests, but until I can reproduce this on my end, I cannot do much about it.

It's alright, we will need to provide the information if we want this fixed :smile: I also can't reproduce this on my laptop with the same sync folders as on my PC where this occurs.

> Also, could I get you to test other file managers? Mainly caja and nautilus. Thanks

No issues in natilus and caja (started from live USB as I wasn't able to start it on my Mint Cinnamon at all).

I can offer you to make a remote support session via VNC or so if you want.

**Comment by denis-beurive** [comment]
Hello,

Just to let you know that I also face the same issue (from this morming).

Nemo crashes systematically.

	dev@dev-VirtualBox:~$ nemo
	Segmentation fault (core dumped)

Information about the system:


	System:    Host: dev-VirtualBox Kernel: 4.15.0-20-generic x86_64 bits: 64 compiler: gcc v: 7.3.0 Desktop: Cinnamon 4.0.10 
	           Distro: Linux Mint 19.1 Tessa base: Ubuntu 18.04 bionic 
	Machine:   Type: Virtualbox System: innotek product: VirtualBox v: 1.2 serial: <filter> 
	           Mobo: Oracle model: VirtualBox v: 1.2 serial: <filter> BIOS: innotek v: VirtualBox date: 12/01/2006 
	Drives:    Local Storage: total: 101.21 GiB used: 28.73 GiB (28.4%) 
	           ID-1: /dev/sda vendor: VirtualBox model: VBOX HARDDISK size: 101.21 GiB 
	Partition: ID-1: / size: 99.12 GiB used: 28.73 GiB (29.0%) fs: ext4 dev: /dev/sda1 
	systemd runlevel: 5 Compilers: gcc: 7.3.0 
	           Shell: bash v: 4.4.19 inxi: 3.0.27 

Information about nemo:

	dev@dev-VirtualBox:~$ nemo --version
	nemo 4.0.6

	dev@dev-VirtualBox:~$ dconf dump /org/nemo/ 
	[window-state]
	start-with-sidebar=false
	geometry='800x438+547+266'
	maximized=false

I can send you the "Core Dump" if you want.

Denis

**Comment by blackdot** [comment]
Still happening to me. Did anybody find at least workaround besides using compact mode?

**Comment by denis-beurive** [comment]
Hello BlackDot,

The only workaround I found is to use Nautilus instead.

However, I made a very strange observation. Let me explain because it is not straightforward to describe what happened.

Everything worked well (with Nemo and Mint in general). Then, one morning, Nemo started to crash systematically. I tried to investigate... but I could not find what was going on. Thus I uninstalled Nemo and installed Nautilus instead.

In the afternoon I rebooted my virtual machine. And I git a message saying that Cinnamon could not open its session. It turned out that cinnamon was not installed anymore ! However, I did not uninstall it (or, at least, not explicitly).

Everything was working well again. Cinnamon started. I could reboot the virtual machine... and I had no problem.

Few days later, I tried to reinstalled Nemo. I was telling me that, maybe something screwed up with my virtual machine, and that may explain the crashes of Nemo. So I reinstalled Nemo, but with no effect whatsoever : it kept crashing systematically.

Thus I uninstalled Nemo again.

Before leaving the office, around 6 p.m., I halted the virtual machine.

An this morning... while I booted the virtual machine, Cinnamon could not start. I turned out that Cinnamon what not installed anymore !

So, I guess that, maybe, somehow, when you uninstall Nemo, you also uninstall Cinnamon (at least on a virtual machine).

**Comment by denis-beurive** [comment]
I made the experiment again, and the scenario is reproducible.

* Uninstall Nemo.
* Reboot.
* Cinnamon cannot start.
* Open a CLI terminal `[Ctrl] + [Alt] + F1`
* Check to see if Cinnamon is installed : `dpkg -l cinnamon`. Cinnamon is not installed.
* Install Cinnamon: `sudo apt-get install cinnamon`
* Reboot : Cinnamon starts. **However Cinnamon does not ask for the user password anymore**.

**Note**: I have taken screenhots that I can give you.

**Comment by tiredofit** [comment]
Happens to me too when opening up in my Home directory with a folder "Nextcloud" in it. Nextcloud client running. Nemo doesn't crash if I open it up from command line outside of that folder.

**Comment by denis-beurive** [comment]
Hello Tiredofit,

Bugs are closely related to (OS) environment. Thus, you should provide data about the environment.

I suggest that you  copy paste the output of the following commands.

```
nemo --version
inxi -Fxz
```

Without these data, any bug report is pretty useless.

Regards

**Comment by tiredofit** [comment]
Version: `nemo 4.0.6`
Inxi:
````bash
System:    Host: ground Kernel: 5.0.10-arch1-1-ARCH x86_64 bits: 64 compiler: gcc v: 8.3.0 Desktop: Cinnamon 4.0.10 
           Distro: Arch Linux 
Machine:   Type: Desktop System: Dell product: XPS 8900 v: N/A serial: <filter> 
           Mobo: Dell model: 0XJ8C4 v: A00 serial: <filter> UEFI [Legacy]: Dell v: 2.3.0 date: 05/16/2018 
CPU:       Topology: Quad Core model: Intel Core i7-6700K bits: 64 type: MT MCP arch: Skylake-S rev: 3 L2 cache: 8192 KiB 
           flags: avx avx2 lm nx pae sse sse2 sse3 sse4_1 sse4_2 ssse3 vmx bogomips: 64152 
           Speed: 4028 MHz min/max: 800/4200 MHz Core speeds (MHz): 1: 4037 2: 4002 3: 4000 4: 4028 5: 4046 6: 4051 7: 4083 
           8: 4014 
Graphics:  Device-1: NVIDIA GM206 [GeForce GTX 960] driver: nvidia v: 418.56 bus ID: 01:00.0 
           Display: x11 server: X.Org 1.20.4 driver: nvidia unloaded: modesetting,vesa 
           resolution: 1920x1080~60Hz, 1920x1080~60Hz, 1920x1080~60Hz 
           OpenGL: renderer: GeForce GTX 960/PCIe/SSE2 v: 4.6.0 NVIDIA 418.56 direct render: Yes 
Audio:     Device-1: Intel 100 Series/C230 Series Family HD Audio vendor: Dell driver: snd_hda_intel v: kernel bus ID: 00:1f.3 
           Device-2: NVIDIA GM206 High Definition Audio driver: snd_hda_intel v: kernel bus ID: 01:00.1 
           Device-3: Logitech H600 [Wireless Headset] type: USB driver: hid-generic,snd-usb-audio,usbhid bus ID: 1-1:7 
           Device-4: Realtek type: USB driver: snd-usb-audio,uvcvideo bus ID: 1-4:3 
           Sound Server: ALSA v: k5.0.10-arch1-1-ARCH 
Network:   Device-1: Intel Ethernet I219-V vendor: Dell driver: e1000e v: 3.2.6-k port: f000 bus ID: 00:1f.6 
           IF: enp0s31f6 state: down mac: <filter> 
           Device-2: Broadcom and subsidiaries BCM4352 802.11ac Wireless Network Adapter vendor: Dell driver: wl v: kernel 
           port: e000 bus ID: 02:00.0 
           IF: wlp2s0 state: up mac: <filter> 
           IF-ID-1: br-27ee62d40f25 state: down mac: <filter> 
           IF-ID-2: br-9b9015092d7a state: down mac: <filter> 
           IF-ID-3: br-9feb12c92e26 state: up speed: N/A duplex: N/A mac: <filter> 
           IF-ID-4: br-a3f46cedb6ea state: down mac: <filter> 
           IF-ID-5: docker0 state: down mac: <filter> 
           IF-ID-6: tun0 state: unknown speed: 10 Mbps duplex: full mac: N/A 
           IF-ID-7: veth4f09cf3 state: up speed: 10000 Mbps duplex: full mac: <filter> 
Drives:    Local Storage: total: 6.94 TiB used: 1.92 TiB (27.7%) 
           ID-1: /dev/sda vendor: Toshiba model: DT01ACA200 size: 1.82 TiB 
           ID-2: /dev/sdb vendor: Transcend model: TS256GSSD320 size: 238.47 GiB 
           ID-3: /dev/sdc vendor: Kingston model: SUV400S37120G size: 111.79 GiB 
           ID-4: /dev/sdd vendor: LITE-ON IT model: L8T-256L9G-11 M.2 2280 256GB size: 238.47 GiB 
           ID-5: /dev/sdf type: USB vendor: Seagate model: Expansion Desk size: 4.55 TiB 
Partition: ID-1: / size: 221.97 GiB used: 101.10 GiB (45.5%) fs: btrfs dev: /dev/sdd3 
           ID-2: /boot size: 487.9 MiB used: 103.5 MiB (21.2%) fs: ext4 dev: /dev/sdd1 
           ID-3: /home size: 221.97 GiB used: 101.10 GiB (45.5%) fs: btrfs dev: /dev/sdd3 
           ID-4: /var size: 221.97 GiB used: 101.10 GiB (45.5%) fs: btrfs dev: /dev/sdd3 
Sensors:   System Temperatures: cpu: 53.5 C mobo: 29.8 C gpu: nvidia temp: 35 C 
           Fan Speeds (RPM): N/A gpu: nvidia fan: 30% 
Info:      Processes: 400 Uptime: 3d 16h 55m Memory: 23.46 GiB used: 14.16 GiB (60.3%) Init: systemd Compilers: gcc: 8.3.0 
           Shell: bash v: 5.0.3 inxi: 3.0.34 
````

This is the output when I navigate to my home directory on anything under than compact:
````bash
Segmentation fault (core dumped)
````

**Comment by mtwebster** [comment]
I added a fix that maybe address this issue, but I still have not been able to reproduce this.  I have a feeling it's caused by invalid timestamps in the file metadata.

In previous versions (and other fm's as well) this specific bit of code would handle it gracefully.  During some optimizations before 4.0.x, this flaw was introduced in https://github.com/linuxmint/nemo/commit/8ec14f032ff927221d3359a873a050954f0221e9.

**Comment by mtwebster** [comment]
If anyone wants to try this out (build it), you can enable some debug options to see if this issue is caught (and for what files):

```
$ NEMO_DEBUG=File G_MESSAGES_DEBUG=all nemo
```

**Comment by tiredofit** [comment]
I've just built and so far so good. Will open a few copies and let it sit for a few hours to see if it fails during some background sync changes.

**Comment by denis-beurive** [comment]
Hello,

I don't know if this can help.

The crashes I reported are happening on a virtual machine.

This weekend, I installed the last version of Linux Mint on a "real box" (the [MintBox 2](https://www.fit-pc.com/web/products/mintbox/mintbox-2/)).

On the "real box", Nemo is working fine.

I give you the configuration of my "real box" :

```
$ inxi -Fxz
System:    Host: labo Kernel: 4.15.0-50-generic x86_64 bits: 64 compiler: gcc v: 7.3.0 
           Desktop: Cinnamon 4.0.10 Distro: Linux Mint 19.1 Tessa base: Ubuntu 18.04 bionic 
Machine:   Type: Laptop System: CompuLab product: Intense-PC v: 1.x serial: <filter> 
           Mobo: CompuLab model: Intense-PC v: 1.x serial: <filter> UEFI [Legacy]: Phoenix 
           v: IPC_2.2.400.1 X64 date: 08/31/2014 
Battery:   Device-1: hidpp_battery_0 model: Logitech Wireless Mouse M215 2nd Gen charge: 100% 
           status: Discharging 
           Device-2: hidpp_battery_1 model: Logitech Wireless Keyboard K330 charge: 100% 
           status: Discharging 
CPU:       Topology: Dual Core model: Intel Core i5-3337U bits: 64 type: MT MCP 
           arch: Ivy Bridge rev: 9 L2 cache: 3072 KiB 
           flags: lm nx pae sse sse2 sse3 sse4_1 sse4_2 ssse3 vmx bogomips: 14368 
           Speed: 2548 MHz min/max: 800/2700 MHz Core speeds (MHz): 1: 2518 2: 2508 3: 2509 
           4: 2510 
Graphics:  Device-1: Intel 3rd Gen Core processor Graphics driver: i915 v: kernel 
           bus ID: 00:02.0 
           Display: x11 server: X.Org 1.19.6 driver: modesetting unloaded: fbdev,vesa 
           resolution: 1680x1050~60Hz 
           OpenGL: renderer: Mesa DRI Intel Ivybridge Mobile v: 4.2 Mesa 18.2.8 
           direct render: Yes 
Audio:     Device-1: Intel 7 Series/C216 Family High Definition Audio driver: snd_hda_intel 
           v: kernel bus ID: 00:1b.0 
           Sound Server: ALSA v: k4.15.0-50-generic 
Network:   Device-1: Intel 82579LM Gigabit Network driver: e1000e v: 3.2.6-k port: 4080 
           bus ID: 00:19.0 
           IF: enp0s25 state: up speed: 1000 Mbps duplex: full mac: <filter> 
           Device-2: Realtek RTL8723AE PCIe Wireless Network Adapter driver: rtl8723ae 
           v: kernel port: 3000 bus ID: 04:00.0 
           IF: wlp4s0 state: down mac: <filter> 
           Device-3: Realtek RTL8111/8168/8411 PCI Express Gigabit Ethernet driver: r8169 
           v: 2.3LK-NAPI port: 2000 bus ID: 05:00.0 
           IF: enp5s0 state: down mac: <filter> 
Drives:    Local Storage: total: 232.89 GiB used: 27.54 GiB (11.8%) 
           ID-1: /dev/sda vendor: Samsung model: SSD 850 EVO 250GB size: 232.89 GiB 
Partition: ID-1: / size: 228.23 GiB used: 27.54 GiB (12.1%) fs: ext4 dev: /dev/sda1 
Sensors:   System Temperatures: cpu: 52.0 C mobo: 52.0 C 
           Fan Speeds (RPM): N/A 
Info:      Processes: 227 Uptime: 26m Memory: 15.12 GiB used: 2.80 GiB (18.5%) Init: systemd 
           runlevel: 5 Compilers: gcc: 7.4.0 Shell: bash v: 4.4.19 inxi: 3.0.27 
```

And:

```
$ nemo --version
nemo 4.0.6
$ cinnamon --version
Cinnamon 4.0.10
```

**Comment by denis-beurive** [comment]
I'll run Nemo on the VM (where Nemo crashes) tomorrow with the setting you give us:

`NEMO_DEBUG=File G_MESSAGES_DEBUG=all nemo`

I'll let you know the result.

Regards.

**Comment by okaestne** [comment]
Build master from source (btw there should be a small how to for noobs like me) and it looks fixed for me using this build. Reinstalled version from repo and the issue occurs again as expected.
Good job!

**Comment by xenopeek** [comment]
Thanks @okaestne, that sounds like it is fixed for Nemo 4.6.4 and Linux Mint 20 Cinnamon. Any still affected by this issue please retest with that version. This issue can be reopened of still reproducible on that version.
